intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py

```diff
@@ -64,93 +64,102 @@ def __init__(self, persist_dir="./output", process=True, input_path=None,
-                                   search_type=self.search_type, search_kwargs=self.search_kwargs)
-    
-    # create a new knowledge base
-    def create(self, input_path, persist_dir):
-        self.doc_parser = DocumentIndexing(retrieval_type=self.retrieval_type, document_store=self.document_store,
-            persist_dir=persist_dir, process=self.process,
-            embedding_model=self.embedding_model, max_length=self.max_length,
-            index_name=self.index_name)
-        self.db = self.doc_parser.KB_construct(input_path)
-        self.retriever = Retriever(retrieval_type=self.retrieval_type, document_store=self.db, 
-                                   top_k=self.top_k, search_type=self.search_type, search_kwargs=self.search_kwargs)
-
-
-    def append_localdb(self, append_path, persist_path):
-        self.db = self.doc_parser.KB_append(append_path, persist_path)
-        self.retriever = Retriever(retrieval_type=self.retrieval_type, document_store=self.db, top_k=self.top_k,
-                           search_type=self.search_type, search_kwargs=self.search_kwargs)
+        knowledge_base = self.database.reload(persist_directory=local_persist_dir, **kwargs)
+        if "retrieval_type" in kwargs:
+            self.retrieval_type = kwargs['retrieval_type']
+        self.retriever = Retriever(retrieval_type=self.retrieval_type, document_store=knowledge_base, **kwargs)
+
+
+    def create(self, input_path, **kwargs):
+        """
+        Create a new knowledge base based on the uploaded files.
+        """
+        data_collection = self.document_parser.load(input=input_path, **kwargs)
+        knowledge_base = self.database.from_collection(content=data_collection, **kwargs)
+        if "retrieval_type" in kwargs:
+            self.retrieval_type = kwargs['retrieval_type']
+        self.retriever = Retriever(retrieval_type=self.retrieval_type, document_store=knowledge_base, **kwargs)
+
+
+    def append_localdb(self, append_path, **kwargs):
+        "Append the knowledge instances into a given knowledge base."
+
+        data_collection = self.document_parser.load(input=append_path, **kwargs)
+        if "retrieval_type" in kwargs:
+            self.retrieval_type = kwargs['retrieval_type']
+        knowledge_base = self.database.from_collection(content=data_collection, **kwargs)
+        self.retriever = Retriever(retrieval_type=self.retrieval_type, document_store=knowledge_base, **kwargs)
 
 
     def pre_llm_inference_actions(self, model_name, query):
         intent = self.intent_detector.intent_detection(model_name, query)
         links = []
         context = ''
-        if self.retriever and self.search_type == "similarity_score_threshold":
+        assert self.retriever is not None, print("Please check the status of retriever")
+        if self.mode == 1:   ## "retrieval with threshold" will only return the document that bigger than the threshold.
```

Comment:
What does the magic number mean?

Reply:
the behaviour of generating the template response. some user may have a higher requirement for the accuracy of the generated content, so we set this magic number to distinguish the RAG behaviour. I will add the introduction in the readme